### PR TITLE
Revert "feat: Implement the bitwise_not in NotExpr (#5902)"

### LINKED ISSUE #6589: Discuss: don't allow not() param is `IntegerType`
### Describe the bug

I'm working for `type coercion`. I find

now
```sql
tests/sqllogictests/test_files/scalar.slt:914

query BB
SELECT not(1), not(0)
----
-2 -1
```

I think we should allow it. it break `type system` and its result is strange.

pg

```sql
psql:commands.sql:15: ERROR:  argument of NOT must be type boolean, not type integer
LINE 1: SELECT not(1), not(0);
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jackwener**
This reverts commit 87e5adc9

# Which issue does this PR close?

Closes #6589

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sqllogictests/test_files/scalar.slt [hunk 1]
```diff
@@ -911,17 +911,15 @@ SELECT not(true), not(false)
 ----
 false true
 
-query BB
+query error DataFusion error: Optimizer rule 'simplify_expressions' failed\ncaused by\nInternal error: NOT 'Literal \{ value: Int64\(1\) \}' can't be evaluated because the expression's type is Int64, not boolean or NULL\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```
Comment (jackwener) [inline datafusion/core/tests/sqllogictests/test_files/scalar.slt:914]: It should fail in `type coercion`, it's a future ticket.
We should try to cast param. (if fail cast then throw a error)

**Comment by jackwener** [comment]
wait for opinion of @liukun4515 in #6589

**Comment by alamb** [comment]
cc @RTEnzyme  who originally contributed this code.

**Comment by alamb** [comment]
Also cc @comphead  who commented on https://github.com/apache/arrow-datafusion/pull/5902

**Review by alamb — APPROVED** [review]

**Comment by RTEnzyme** [comment]
I agree that `not()` should only support boolean types. If we want to implement bitwise_not on integers, we should create a new Expr.

**Pull request merged**